### PR TITLE
RegisterGameDataObjects Function

### PR DESCRIPTION
**Pull request by Yariazen**
Adds IAddGDO interface used to tag custom gdos.

Adds RegisterGameDataObjects(Type t) which when passed a mod's type, will register all of their game data objects marked with IAddGDO.

Adds AddGameDataObject(Type t) for adding game data objects based on a type.

### CODE REVIEW

**Comment by Yariazen** [comment]
Example use is 
`
RegisterGameDataObjects(typeof(Mod));
`

File: KitchenLib/src/BaseMod.cs [hunk 1]
```diff
@@ -322,5 +345,19 @@ public KitchenLogger InitLogger()
...
+			Assembly asm = ModRegistery.keyValuePairs[t];
+			foreach (Type type in asm.GetTypes())
+			{
+				if(typeof(IAddGDO).IsAssignableFrom(type))
+				{
+					keyValuePairs.Add(type.Name, AddGameDataObject(type));
+				}
+			}
+			return keyValuePairs;
+		}
```
Comment (StarFluxGames) [inline KitchenLib/src/BaseMod.cs:361]: Not sure if this is intentional, but this method doesn't actually register anything?

**Comment by Yariazen** [comment]
Btw this can be closed now right? The feature is already in the beta kl.

**Comment by StarFluxGames** [comment]
> Btw this can be closed now right? The feature is already in the beta kl.

Yep you're right

**Pull request closed without merging**